### PR TITLE
feat: implement Mandrill provider connection test

### PR DESCRIPTION
**Pull request by daniel-shuy**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
    - Fix Mandrill provider test (`mandrill.provider.spec.ts`): The test was mocking `MandrillProvider.sendMessage`, the function being tested itself, which defeats the purpose of the test. Fixed by mocking `MandrillProvider.transporter.messages.send` instead.
    - Implement `MandrillProvider.checkIntegration` so that users can test their credentials when configuring a Mandrill provider.

- **Why was this change needed?** (You can also link to an open issue here)
#1467

- **Other information**:
Mailchimp Transactional provides a [/users/ping](https://mailchimp.com/developer/transactional/api/users/ping/) endpoint to test API keys (https://mailchimp.com/developer/transactional/guides/quick-start/#make-your-first-api-call).

### CODE REVIEW

File: providers/mandrill/src/lib/mandrill.provider.ts [hunk 1]
```diff
@@ -64,13 +64,21 @@ export class MandrillProvider implements IEmailProvider {
...
+
+      return {
+        success: true,
+        message: 'Integrated successfully!',
+        code: CheckIntegrationResponseEnum.SUCCESS,
+      };
+    } catch (error) {
+      return {
+        success: false,
+        message: error?.message,
```
Comment (BiswaViraj) [inline providers/mandrill/src/lib/mandrill.provider.ts:79]: can we have better error mapping, similar to the one you did in the previous PR?

Comment (daniel-shuy) [reply providers/mandrill/src/lib/mandrill.provider.ts:79]: Unfortunately I searched through the entire API documentation (https://mailchimp.com/developer/transactional/api/) and [mailchimp-transactional](https://github.com/mailchimp/mailchimp-transactional-node/blob/master/src/ApiClient.js#L70-L104) source code (it is simply using [axios](https://axios-http.com/)), and could not find any error code mapping

Comment (BiswaViraj) [reply providers/mandrill/src/lib/mandrill.provider.ts:79]: https://mailchimp.com/developer/marketing/docs/errors/#error-glossary
is this helpful?
i think 401 can be mapped

Comment (daniel-shuy) [reply providers/mandrill/src/lib/mandrill.provider.ts:79]: That documentation is for the Marketing API though, not the Transactional Email API, they are 2 totally different products

Comment (BiswaViraj) [reply providers/mandrill/src/lib/mandrill.provider.ts:79]: ok thought so,
Alright then let's leave it as it is
Thank you 🚀

**Review by BiswaViraj — APPROVED** [review]

**Pull request merged**